### PR TITLE
Add Validation Of Search Attributes For The Scheduled Workflow

### PR DESCRIPTION
**Pull request by ast2023**
## What changed?
Scheduled workflow search attributes validated on CreateSchedule request.

## Why?
User Request.

## How did you test it?
Run tests.

## Potential risks
Unknown.

## Is hotfix candidate?
No

### CODE REVIEW

File: service/frontend/workflow_handler.go [hunk 1]
```diff
@@ -2631,6 +2637,27 @@ func (wh *WorkflowHandler) CreateSchedule(ctx context.Context, request *workflow
...
+	action, ok := scheduledAction.(*schedpb.ScheduleAction_StartWorkflow)
+	if !ok {
+		panic(scheduledAction)
+	}
+	searchAttributes := action.StartWorkflow.SearchAttributes
+	unaliased, err := searchattribute.UnaliasFields(wh.saMapperProvider, searchAttributes, namespaceName.String())
+	if err != nil {
+		return err
+	}
+	return wh.validateSearchAttributes(unaliased, namespaceName)
```
Comment (dnr) [inline service/frontend/workflow_handler.go:2658]: this is already done here: https://github.com/temporalio/temporal/blob/main/service/frontend/workflow_handler.go#L2679-L2683 inside validateStartWorkflowArgsForSchedule

File: service/frontend/workflow_handler.go [hunk 2]
```diff
@@ -2631,6 +2637,27 @@ func (wh *WorkflowHandler) CreateSchedule(ctx context.Context, request *workflow
...
+	namespaceName namespace.Name,
+) error {
+
+	scheduledAction := request.GetSchedule().GetAction().GetAction()
+	if scheduledAction == nil {
+		panic("no scheduled workflow")
+	}
+	action, ok := scheduledAction.(*schedpb.ScheduleAction_StartWorkflow)
+	if !ok {
+		panic(scheduledAction)
```
Comment (dnr) [inline service/frontend/workflow_handler.go:2651]: there may be other action types in the future, you can't panic here

Comment (tdeebswihart) [reply service/frontend/workflow_handler.go:2651]: We should switch on the type and return an error in the `default` case, no?

File: service/frontend/workflow_handler.go [hunk 3]
```diff
@@ -2600,6 +2600,12 @@ func (wh *WorkflowHandler) CreateSchedule(ctx context.Context, request *workflow
 			ConflictToken: scheduler.InitialConflictToken,
 		},
 	}
+
+	err = wh.validateScheduledWorkflowSearchAttributes(request, namespaceName)
+	if err != nil {
+		return nil, fmt.Errorf("validating scheduled workflow search attributes:%w", err)
```
Comment (tdeebswihart) [inline service/frontend/workflow_handler.go:2606]:
```suggestion
		return nil, fmt.Errorf("validating scheduled workflow search attributes: %w", err)
```

File: service/frontend/workflow_handler.go [hunk 4]
```diff
@@ -2631,6 +2637,27 @@ func (wh *WorkflowHandler) CreateSchedule(ctx context.Context, request *workflow
...
 }
 
+func (wh *WorkflowHandler) validateScheduledWorkflowSearchAttributes(
+	request *workflowservice.CreateScheduleRequest,
+	namespaceName namespace.Name,
+) error {
+
+	scheduledAction := request.GetSchedule().GetAction().GetAction()
+	if scheduledAction == nil {
+		panic("no scheduled workflow")
```
Comment (tdeebswihart) [inline service/frontend/workflow_handler.go:2647]: @dnr are we ok with panicking here?

**Pull request closed without merging**